### ISSUE TITLE
Video views are not adding up after upgrade

### ISSUE THREAD

**Issue by Mike35538** [Chocobozzz/PeerTube#5367]
### Describe the current behavior

After a recent upgrade 4.2.2 to 4.30 plus migrating to another server my view counts aren't adding up.

I get this error a lot now and the view counts are smaller than normal.

warn[10/18/2022, 2:50:26 PM] Too much watch section to store for a viewer, skipping this one
{
  "currentTime": 2611,
  "tags": [
    "views",
    "9bd34395d-33e8-42a6-947b-d8e0a850abd2"
  ]
}

Thanks

### Additional information

Node v18.11.0

**Comment by Mike35538** [comment]
Hi kontrollanten, thanks for the help. I see my server's ip address where it says 'PeerTube thinks your web browser public IP'. Hmmm... I edited the 127.0.0.1 address in to my server ip address in the production.yaml and /etc/nginx/sites-available/peertube config. I did that because I kept getting a upstream error and peertube showed a gateway 502 error until i put my server ip in those places. I'm not on a CDN or reverse proxy.

**Comment by Mike35538** [comment]
It now says PeerTube thinks your web browser public IP is 127.0.0.1.
In nginx config now i have upstream backend {server 127.0.0.1:9000;}
In production.yaml now I have the hostname: '127.0.0.0' and the trust_proxy: -'my server ip address'

I'm not getting the error 'warn[10/18/2022, 2:50:26 PM] Too much watch section to store for a viewer, skipping this one' as show above anymore, now it is showing alot of errors like this:
warn[10/18/2022, 4:29:25 PM] Invalid end watch section 216. Last start record was at 284.
warn[10/18/2022, 4:29:30 PM] Invalid end watch section 221. Last start record was at 284. 

I feel like this started when I attempted to upgrade from PT 4.2.2 to 4.30, the upgrade aborted because of this message:
[3/5] Fetching packages...
error @uploadx/core@6.0.0: The engine "node" is incompatible with this module. Expected version ">=14.18.20". Got "14.18.1"
error Found incompatible module.

So then I upgraded Node to v18.11.0 and tried the upgrade script again and it all went well. PeerTube wouldn't load however and I got a 502 bad gateway. The journalctl -fu Peertube logs showed an upstream error, I forgot what it said exactly. A little research led me to believe I needed to put in my server ip in upstream backend for nginx config. That made the PeerTube load and the 502 error go away. But now I have trouble with the views being counted incorrectly.

Thanks for any help.
